scene-recognition/process.py

```diff
@@ -38,46 +53,99 @@ def format_timecode(value):
-        destination = golds_dir / file.with_suffix('.csv').name
-        # read in as dataframe to more easily manipulate columns
-        df = pd.read_csv(source)
-        # create new timestamp column and fill with values
-        df.insert(1, 'at', "")
-        df['at'] = df['filename'].apply(format_timecode)
-        # remove unseen
-        df = df[(df.seen != "false") & (df.seen != "False")]
-        # remove seen column
-        df = df.drop('seen', axis=1)
-        # any that are left have been seen. therefore, any rows with label = "" are negative
-        # so their labels should be changed to "-"
-        df.loc[df['type label'].isna(), 'type label'] = '-'
-        # then rename columns
-        df = df.rename(columns={
+def copy_timepoints(source, destination):
+    """Copy time points to golds directory after some changes."""
+    df = pd.read_csv(source)
+    df = amend_dataframe(df)
+    # output to csv
+    df.to_csv(destination, index=False)
+
+
+def amend_dataframe(df):
+    # create new timestamp column and fill with values
+    df.insert(1, 'at', "")
+    df['at'] = df['filename'].apply(format_timecode)
+    # Remove unseen rows and the seen column
+    # Note: as of April 2025 this does not appear to ever happen
+    df = df[(df.seen != "false") & (df.seen != "False")]
+    df = df.drop('seen', axis=1)
+    # Any rows that are left have been seen, therefore, any rows with label = "" are
+    # negative so their labels should be changed to "-"
+    df.loc[df['type label'].isna(), 'type label'] = '-'
+    df = df.rename(columns={
             'type label': 'scene-type',
             'subtype label': 'scene-subtype',
-            'modifier': 'transitional',
-        })
-        # remove first column (filename)
-        df = df.drop('filename', axis=1)
-        # remove transcript and note columns
-        df = df.drop('transcript', axis=1)
-        df = df.drop('note', axis=1)
-        # output to csv with same filename
-        df.to_csv(destination, index=False)
+            'modifier': 'transitional'})
+    # remove filename, transcript and note columns
+    df = df.drop('filename', axis=1)
+    df = df.drop('transcript', axis=1)
+    df = df.drop('note', axis=1)
+    return df
 
 
-if __name__ == '__main__':
-    task_dir = pathlib.Path(__file__).parent
-    golds_dir = task_dir / 'golds'
+def copy_timeframes(source, destination):
+    """This was used at some point to create timeframes from timepoints, but we are not
+    storing derivable data anymore. Keeping it around for a while in case it is useful,
+    but this function will be removed at some point."""
+    df = pd.read_csv(source)
+    df = amend_dataframe(df)
+    current_label = None
+    current_sublabel = None
+    current_frame = []
+    with open(destination, "w") as csv_file:
+        csv_writer = csv.writer(csv_file)
+        csv_writer.writerow(_get_column_names())
+        for index, row in df.iterrows():
+            ts = row['timestamp']
```

Comment:
from here to following several lines, field names are not correct.

Reply:
Fixed